cmd/kubeadm/app/preflight/checks.go

```diff
@@ -431,8 +449,13 @@ func (sysver SystemVerificationCheck) Check() (warnings, errors []error) {
 	var warns []error
 	// All the common validators we'd like to run:
 	var validators = []system.Validator{
-		&system.KernelValidator{Reporter: reporter},
-		&system.DockerValidator{Reporter: reporter}}
+		&system.KernelValidator{Reporter: reporter}}
+
+	// run the docker validator only with dockershim
+	if sysver.CRISocket == "/var/run/dockershim.sock" {
+		// TODO(runcom): move the DockerValidator to actually use the CRI.Version endpoint
```

Comment:
can you create an issue to follow this up please?

Reply:
created https://github.com/kubernetes/kubeadm/issues/533